utils/utils.go

```diff
@@ -266,6 +266,15 @@ func IsAddressLegal(address string) bool {
 	return isLegal
 }
 
+func CheckIfPortAvailable(port int) error {
```

Comment:
Please add doc comment, can we also add tests?

Reply:
I think for test we will have to depend on some random port which might be occupied in some cases and test will be error prone. thoughts?
Also, this method just calls the `net.Listen` and does not have much business logic.

Reply:
We can `net.Listen` on the same port before calling this function to test. The port can be fixed or generated from freeport pkg. 

Ok with skipping the test, but it's good to have. /cc @mukundansundar

Reply:
As discussed, this test can be tracked separately.

Reply:
There is no E2E for dashboard. We can try to add one and cover this there.
tracking issue - https://github.com/dapr/cli/issues/1192

Reply:
Thanks @pravinpushkar